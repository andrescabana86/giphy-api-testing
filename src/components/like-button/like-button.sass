.like-button
  align-items: center
  background-color: white
  border: 0
  color: black
  display: flex
  font-weight: bolder
  justify-content: center
  margin: 0
  padding: 0
  &__icon
    display: block
    margin: 0 4px
    width: 14px