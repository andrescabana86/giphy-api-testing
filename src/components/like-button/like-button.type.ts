export type TLikeButtonProps = {
  active: boolean
  onClick?(): void
  text?: string
}
