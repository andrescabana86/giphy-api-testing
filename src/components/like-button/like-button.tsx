import React from 'react'
import { TLikeButtonProps } from '@components/like-button/like-button.type'
import './like-button.sass'

export const LikeButton = React.memo((props: TLikeButtonProps) => {
  const { active, text, onClick = undefined } = props
  return (
    <button
      className={`like-button${active ? ' like-button--active' : ''}`}
      data-testid="like-button"
      onClick={onClick}
    >
      {text?.length ? text : active ? 'Liked!' : 'Like'}
      <img
        alt="Like"
        className="like-button__icon"
        src={props.active ? 'static/icons/heart-red.svg' : 'static/icons/heart-empty-red.svg'}
      />
    </button>
  )
})
