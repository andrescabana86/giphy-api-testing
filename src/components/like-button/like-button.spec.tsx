import React from 'react'
import { render, cleanup } from '@testing-library/react'
import { LikeButton } from '@components/like-button/like-button'

describe('LikeButton', () => {
  afterEach(cleanup)

  it('should render LikeButton component', () => {
    // Arrange
    // Act
    const { asFragment } = render(<LikeButton active={false} />)

    // Assert
    expect(asFragment()).toMatchSnapshot()
  })

  it('should show text "Like" when button has non-active state"', () => {
    // Arrange
    // Act
    const { getByTestId } = render(<LikeButton active={false} />)
    const button = getByTestId('like-button')

    // Assert
    expect(button).toHaveTextContent('Like')
    expect(button).toHaveClass('like-button')
    expect(button).not.toHaveClass('like-button--active')
  })

  it('should show text "Liked!" when button has active state"', () => {
    // Arrange
    // Act
    const { getByTestId } = render(<LikeButton active={true} />)
    const button = getByTestId('like-button')

    // Assert
    expect(button).toHaveTextContent('Liked!')
    expect(button).toHaveClass('like-button')
    expect(button).toHaveClass('like-button--active')
  })

  it('should show custom "Love it!" text when receives text by props"', () => {
    // Arrange
    // Act
    const customText = 'Love it!'
    const { getByTestId } = render(<LikeButton active={true} text={customText} />)
    const button = getByTestId('like-button')

    // Assert
    expect(button).toHaveTextContent(customText)
    expect(button).toHaveClass('like-button', 'like-button--active')
  })
})
