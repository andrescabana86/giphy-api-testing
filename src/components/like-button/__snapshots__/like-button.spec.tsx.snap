// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LikeButton should render LikeButton component 1`] = `
<DocumentFragment>
  <button
    class="like-button"
    data-testid="like-button"
  >
    Like
    <img
      alt="Like"
      class="like-button__icon"
      src="static/icons/heart-empty-red.svg"
    />
  </button>
</DocumentFragment>
`;
